utils/gddrescue/Makefile

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (C) 2018 Lucian Cristian
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gddrescue
+PKG_VERSION:=1.22
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)_$(PKG_VERSION).orig.tar.xz
+PKG_SOURCE_URL:=http://http.debian.net/debian/pool/main/g/$(PKG_NAME)
+PKG_HASH:=a715488f77ae201099ff99360baf7d909bce0fa6c43588e076e830fec93d09f1
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+
+PKG_LICENSE:=GPL-2.0
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/gddrescue
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Data recovery tool
+  URL:=https://www.gnu.org/software/ddrescue/
+  DEPENDS:= +libstdcpp
```

Comment:
is uclibc++ possible here?

Reply:
I'm getting this

```
In file included from /home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/string:25:0,
                 from mapfile.cc:26:
/home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/vector: In instantiation of 'void std::vector<T, Allocator>::_insert_from_iterator(std::vector<T, Allocator>::iterator, InputIterator, InputIterator) [with InputIterator = Sblock*; T = Sblock; Allocator = std::allocator<Sblock>; std::vector<T, Allocator>::iterator = Sblock*]':
/home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/vector:288:26:   required from 'void std::vector<T, Allocator>::_dispatch_insert(std::vector<T, Allocator>::iterator, InputIterator, InputIterator, std::__false_type) [with InputIterator = Sblock*; T = Sblock; Allocator = std::allocator<Sblock>; std::vector<T, Allocator>::iterator = Sblock*]'
/home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/vector:297:20:   required from 'void std::vector<T, Allocator>::insert(std::vector<T, Allocator>::iterator, InputIterator, InputIterator) [with InputIterator = Sblock*; T = Sblock; Allocator = std::allocator<Sblock>; std::vector<T, Allocator>::iterator = Sblock*]'
mapfile.cc:331:77:   required from here
/home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/vector:270:6: error: no matching function for call to 'Sblock::Sblock()'
    T temp;
      ^~~~
In file included from mapfile.cc:30:0:
block.h:107:3: note: candidate: Sblock::Sblock(long long int, long long int, Sblock::Status)
   Sblock( const long long p, const long long s, const Status st )
   ^~~~~~
block.h:107:3: note:   candidate expects 3 arguments, 0 provided
block.h:105:3: note: candidate: Sblock::Sblock(const Block&, Sblock::Status)
   Sblock( const Block & b, const Status st )
   ^~~~~~
block.h:105:3: note:   candidate expects 2 arguments, 0 provided
block.h:95:7: note: candidate: Sblock::Sblock(const Sblock&)
 class Sblock : public Block
       ^~~~~~
block.h:95:7: note:   candidate expects 1 argument, 0 provided
Makefile:64: recipe for target 'mapfile.o' failed
make[4]: *** [mapfile.o] Error 1
```

Reply:
The error here is due to uClibc++ not supporting long long. Sigh.

Reply:
Upstream provided a patch which will be part of 1.25 that fixes compilation with uclibc++. It's really quite simple.

https://gist.github.com/neheb/5e340598e2e9c31939a3570d106816aa